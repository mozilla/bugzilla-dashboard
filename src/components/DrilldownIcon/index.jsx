import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import ExpandMore from '@material-ui/icons/ExpandMore';

const styles = ({
  svgWrapper: {
    '&:hover': {
      backgroundColor: 'lightGray',
    },
  },
  icon: {
    fontSize: '1rem',
    verticalAlign: 'bottom',
  },
});

const DrilldownIcon = ({
  classes, name, onChange, properties,
}) => (
  <div
    className={classes.svgWrapper}
    name={name}
    onKeyPress={e => onChange(e, properties)}
    onClick={e => onChange(e, properties)}
    role="button"
    tabIndex="0"
  >
    <ExpandMore classes={{ root: classes.icon }} />
  </div>
);

DrilldownIcon.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  properties: PropTypes.shape({}).isRequired,
};

export default withStyles(styles)(DrilldownIcon);
