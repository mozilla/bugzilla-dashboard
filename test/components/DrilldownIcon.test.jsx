import React from 'react';
import renderer from 'react-test-renderer';
import DrilldownIcon from '../../src/components/DrilldownIcon';

it('renders the details for a Bugzilla component', () => {
  const tree = renderer
    .create((
      <DrilldownIcon
        name="x"
        properties={{ y: 'z' }}
        onChange={() => null}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
