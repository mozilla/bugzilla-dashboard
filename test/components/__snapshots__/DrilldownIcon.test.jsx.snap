// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the details for a Bugzilla component 1`] = `
<div
  className="DrilldownIcon-svgWrapper-1"
  name="x"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
  y="z"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root-3 DrilldownIcon-icon-2"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</div>
`;
